cubic-server/Player.cpp

```diff
@@ -942,6 +953,14 @@ void Player::_onSetPlayerPositionAndRotation(protocol::SetPlayerPositionAndRotat
     onEvent(Server::getInstance()->getPluginManager(), onEntityRotate, this, {_rot.x, _rot.z, 0}, {(uint8_t) pck.yaw, (uint8_t) pck.pitch, 0});
     this->setPosition(pck.x, pck.feetY, pck.z, pck.onGround);
     this->setRotation(pck.yaw, pck.pitch);
+    if (Entity::pickupItem().first) {
+        this->sendPickupItem({Entity::pickupItem().second->getId(), this->getId(), Entity::getPickupItemFromEntity(Entity::pickupItem().second).second});
+        auto slotItem =
+            protocol::Slot {true, Entity::getPickupItemFromEntity(Entity::pickupItem().second).first, Entity::getPickupItemFromEntity(Entity::pickupItem().second).second};
+        this->_inventory->insert(slotItem);
+        this->sendSetContainerContent({_inventory});
+        this->getDimension()->removeEntity(Entity::pickupItem().second->getId());
+    }
```

Comment:
Why is this different than before ?
Can't you just put that in a separate function ?